lib/combo.coffee

```diff
@@ -667,6 +664,20 @@
     updateClassNames: () ->
       if @classNameOnEmpty
         @input.toggleClass @classNameOnEmpty, not @getRawValue()
+                
+    createLabel: (item) ->
+      label = @label?(item,  @getRawValue())
+      if label? then "<span class='#{label.className}'>#{label.text}</span>" else ""
+    
+    updateLabel: () ->
+      if @inputLabel? and (@isExpanded or @getRawValue() == "" or @createLabel(@getSelectedItem()) == "")
+          @inputLabel.remove() 
+          @inputLabel = null
+          @el.removeClass('has-label')
+      else if @inputLabel == null and !@isExpanded and @createLabel(@getSelectedItem()) != ""

```

Comment:
I hvilke tilfælde havner vi ikke i hverken den første eller den anden clausul?
Du må meget gerne prøve at reducere de to udtryk lidt eller bryde det op i flere.

@createLabel(@getSelectedItem()) == "" bliver kaldt tre steder, så du kan med fordel bruge en variabel her.

burde den ikke hedde updateInputLabel?